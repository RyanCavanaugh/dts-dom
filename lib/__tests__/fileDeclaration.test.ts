import {create, DeclarationFlags, emit, type} from "../index"

describe("a file declaration", () => {
    it("matches the snapshot", () => {
        const file = create.file('MyFile.d.ts')

        expect(emit(file)).toMatchSnapshot();
    });

    it("with imports and members", () => {
      const importName = 'MyNamedImport'
      const file = create.file()

      file.imports.push(create.importNamed(importName, '', importName))
      file.members.push(create.const('MY_CONST', create.interface(importName), DeclarationFlags.Export))
      file.members.push(create.const('YOUR_CONST', type.stringLiteral('Literally'), DeclarationFlags.Export))

      expect(emit(file)).toMatchSnapshot();
  });
});
