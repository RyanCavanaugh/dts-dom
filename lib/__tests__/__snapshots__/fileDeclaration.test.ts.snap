// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a file declaration matches the snapshot 1`] = `
// Filename: MyFile.d.ts


`;

exports[`a file declaration with imports and members 1`] = `
import {MyNamedImport} from 'MyNamedImport';

export const MY_CONST: MyNamedImport;

export const YOUR_CONST: "Literally";


`;
